Sources/SparkConnect/DataFrame.swift

```diff
@@ -153,16 +153,35 @@ public actor DataFrame: Sendable {
             let arrowResult = ArrowReader.makeArrowReaderResult()
             _ = reader.fromMessage(schema, dataBody: Data(), result: arrowResult)
             _ = reader.fromMessage(dataHeader, dataBody: dataBody, result: arrowResult)
-            await self.addBathes(arrowResult.batches)
+            await self.addBatches(arrowResult.batches)
           }
         }
       }
     }
   }
 
-  /// This is designed not to support this feature in order to simplify the Swift client.
-  public func collect() async throws {
-    throw SparkConnectError.UnsupportedOperationException
+  /// Execute the plan and return the result as ``[[String?]]``.
+  /// - Returns: ``[[String?]]``
+  public func collect() async throws -> [[String?]] {
+    try await execute()
+
+    var result: [[String?]] = []
+    for batch in self.batches {
+      for i in 0..<batch.length {
+        var values: [String?] = []
+        for column in batch.columns {
+          let str = column.array as! AsString
```

Comment:
For DataFrame, I suppose that the return of `collect` is an array of `Row`. But this `collect` returns strings. Is it just for initial implementation and will be `Row` later?

Reply:
Yes, correct. `Row` implementation is on the way~

Reply:
For `Scala` client, we also support `Array[Long]` like the following.
```
$ bin/spark-shell --remote sc://localhost:15002
scala> spark.range(1).collect()
res0: Array[java.lang.Long] = Array(0L)
```